pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -200,37 +201,35 @@ func (a *GCPActuator) getImageID(cd *hivev1.ClusterDeployment, logger log.FieldL
 // for use in the name of the machine pool. We are severely restricted on name lengths on GCP
 // and effectively have one character of flexibility with the naming convention originating in
 // the installer.
-func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDeployment, logger log.FieldLogger) (string, bool, error) {
+func (a *GCPActuator) obtainLease(pool *hivev1.MachinePool, cd *hivev1.ClusterDeployment, logger log.FieldLogger) (leaseChar string, leaseCreated bool, leaseErr error) {
 	logger.Debugf("obtaining lease for pool: %s", pool.Name)
 
 	leases := &hivev1.MachinePoolNameLeaseList{}
 	err := a.client.List(context.TODO(), leases, client.InNamespace(pool.Namespace),
-		client.MatchingLabels(map[string]string{constants.ClusterDeploymentNameLabel: cd.Name}))
+		client.MatchingLabels(map[string]string{
+			constants.ClusterDeploymentNameLabel: cd.Name,
+			constants.MachinePoolNameLabel:       pool.Name,
+		}))
 	if err != nil {
 		return "", false, err
 	}
-	logger.Debugf("found %d leases in namespace", len(leases.Items))
-	for _, l := range leases.Items {
-		logger.Debugf("checking lease: %s", l.Name)
-		for _, ownerRef := range l.OwnerReferences {
-			logger.Debugf("checking owner ref: %v", ownerRef)
-			if ownerRef.APIVersion == "hive.openshift.io/v1" &&
-				ownerRef.Kind == "MachinePool" &&
-				ownerRef.Name == pool.Name {
-
-				logger.Debugf("machine pool already has lease: %s", l.Name)
-
-				// Ensure the lease name is in the format we expect, we know everything up to
-				// the last character.
-				expectedLeasePrefix := fmt.Sprintf("%s-", cd.Spec.ClusterMetadata.InfraID)
-				if l.Name[0:len(l.Name)-1] != expectedLeasePrefix {
-					return "", false, fmt.Errorf("lease %s did not match expected lease name format (%s[CHAR])", l.Name, expectedLeasePrefix)
-				}
-
-				return string(l.Name[len(l.Name)-1]), false, nil
-			}
+	logger.Debugf("found %d leases for pool", len(leases.Items))
+	if len(leases.Items) > 1 {
+		return "", false, fmt.Errorf("found %d leases for pool, manual intervention required", len(leases.Items))
+	} else if len(leases.Items) == 1 {
+		l := leases.Items[0]
+		logger.Debugf("machine pool already has lease: %s", l.Name)
+
+		// Ensure the lease name is in the format we expect, we know everything up to
+		// the last character.
+		expectedLeasePrefix := fmt.Sprintf("%s-", cd.Spec.ClusterMetadata.InfraID)
```

Comment:
I think this can be made a little easier to read in the following way. Take it or leave it. Your choice.

```
leaseChar := l.Name[len(l.Name)-1:]
expectedLeaseName := fmt.Sprintf("%s-%s", cd.Spec.ClusterMetadata.InfraID, leaseChar)
if expectedLeastName != l.Name {
  return "", false, fmt.Errorf(...)
}
return leaseChar, false, nil
```